java/util/HashMap/HashMap.hpp

```diff
@@ -24,502 +24,760 @@
-									break;
-								case '\f':
-									replacementString = R"(\f)";
-									break;
-								case '\n':
-									replacementString = R"(\n)";
-									break;
-								case '\r':
-									replacementString = R"(\r)";
-									break;
-								case '\t':
-									replacementString = R"(\t)";
-									break;
-								case '\\':
-									replacementString = R"(\\)";
-									break;
-								default:
-										string charAtIndexString = string_from_char(charAtIndex);
-										replacementString = charAtIndexString;
-										free(charAtIndexString);
-							}
-
-							result += replacementString;
-							index++;
-						}
-
-						return result;
+	namespace Util {
+		template <class Key, class Value>
+		class HashMap :
+			public AbstractMap,
+			public virtual Map<Key, Value>,
+			public virtual Cloneable,
+			public virtual Serializable {
+
+		private:
+			std::map<Key, Value> original;
+			String backup;
+			typedef typename std::map<Key, Value>::iterator IteratorType;
+			typedef typename std::map<Key, Value>::const_iterator ConstIteratorType;
+
+		public:
+			/**
+			 * Constructs an empty HashMap
+			 */
+			HashMap() {
+			}
+
+			/**
+			 * Constructs a new HashMap with the same
+			 * mappings as the specified Map.
+			 *
+			 * @param target  	the map whose mappings
+			 * 					are to be placed in this map
+			 */
+			HashMap(const HashMap<Key, Value> &target) {
+				this->putAll(target);
+			}
+
+			/**
+			 * Destructor HashMap
+			 */
+			~HashMap() {
+			}
+
+		public:
+            /**
+             * @return begin iterator of this object
+             */
+			IteratorType begin() {
+				return this->original.begin();
+			}
+
+            /**
+             * @return const begin iterator of this object
+             */
+			ConstIteratorType begin() const {
+				return this->original.begin();
+			}
+
+            /**
+             * @return end iterator of this object
+             */
+			IteratorType end() {
+				return this->original.end();
+			}
+
+            /**
+             * @return const end iterator of this object
+             */
+			ConstIteratorType end() const {
+				return this->original.end();
+			}
+		public:
+			/**
+			 * Compares the specified object
+			 * with this map for equality.
+			 *
+			 * @param object
+			 */
+			boolean equals(const HashMap<Key, Value> &anotherHashMap)  {
+
+                if (this->size() != anotherHashMap.size()) {
+                    return false;
+                }
+
+                for (auto const &thisElement : this->original) {
+                    Value anotherValue = anotherHashMap.get(thisElement.first);
+
+                    if (thisElement.second != anotherValue) {
+                        return false;
+                    }
+                }
+
+                return true;
+			}
+
+			/**
+			 * forEach method: loop a HashMap and do the function action
+			 *
+			 * @param action
+			 */
+			void forEach (BiConsumer<void(Key, Value)> action) {

```

Comment:
I think we should comment this and wait for BiConsumer class

Reply:
Thanks. Fixed.